tests/phpunit/test-class-wc-stripe-upe-payment-gateway.php

```diff
@@ -1537,6 +1537,9 @@ public function test_if_order_has_subscription_payment_method_will_be_saved() {
 
 		list( $amount, $description, $metadata ) = $this->get_order_details( $order );
 
+		// When the order contains a subscription, the payment type is expected to be "recurring".
+		$metadata['payment_type'] = 'recurring';
```

Comment:
This test isn't related to the introduced changes. It started to fail in [f9cf21f](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3139/commits/f9cf21f574d5c6e88cf53b073e395ccf4023fcf4) due to the introduction of the [WC_Subscriptions test helper class](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3139/commits/f9cf21f574d5c6e88cf53b073e395ccf4023fcf4#diff-5c210fbd4f0b20f01c9cd83dd44302bc82a332b9e22f4cbe72e28f991192b883).

We check whether the payment is recurring [when setting up](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.3.0/includes/payment-methods/class-wc-stripe-upe-payment-gateway.php#L1694) this metadata, calling [is_payment_recurring()](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.3.0/includes/compat/trait-wc-stripe-subscriptions-utilities.php#L52-L66). Before creating the test helper class, [is_subscriptions_enabled()](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/8.3.0/includes/compat/trait-wc-stripe-subscriptions-utilities.php#L14-L23) likely returned `false` due to the `class_exists( 'WC_Subscriptions' )` check. This would make the previous assertion pass, even though it's not what would be expected.